File: app/src/main/java/com/example/schoolairdroprefactoredition/domain/DomainResult.java
package com.example.schoolairdroprefactoredition.domain;

public class DomainResult {

    /**
     * success : true
     */
    private int code;

    private long time;

    private String msg;

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public long getTime() {
        return time;
    }

    public void setTime(long time) {
        this.time = time;
    }

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    public boolean isSuccess() {
        return code == 200;
    }
}

File: app/src/main/java/com/example/schoolairdroprefactoredition/domain/DomainResult.kt
package com.example.schoolairdroprefactoredition.domain

import com.example.schoolairdroprefactoredition.utils.ConstantUtil
import java.io.Serializable

data class DomainResult(
        val code: Int,
        val msg: String) : Serializable {

    val isSuccess
        get() = code == ConstantUtil.HTTP_OK
}